testing/path.go

```diff
@@ -39,47 +41,57 @@ func (path *Path) SetChannelOrdered() {
 // if EndpointA does not contain a packet commitment for that packet. An error is returned
 // if a relay step fails or the packet commitment does not exist on either endpoint.
 func (path *Path) RelayPacket(packet channeltypes.Packet) error {
+	_, _, err := path.RelayPacketWithResults(packet)
+	return err
+}
+
+// RelayPacketWithResults attempts to relay the packet first on EndpointA and then on EndpointB
+// if EndpointA does not contain a packet commitment for that packet. The function returns:
+// - The result of the packet receive transaction.
+// - The acknowledgement written on the receiving chain.
+// - An error if a relay step fails or the packet commitment does not exist on either endpoint.
+func (path *Path) RelayPacketWithResults(packet channeltypes.Packet) (*sdk.Result, []byte, error) {
 	pc := path.EndpointA.Chain.App.GetIBCKeeper().ChannelKeeper.GetPacketCommitment(path.EndpointA.Chain.GetContext(), packet.GetSourcePort(), packet.GetSourceChannel(), packet.GetSequence())
 	if bytes.Equal(pc, channeltypes.CommitPacket(path.EndpointA.Chain.App.AppCodec(), packet)) {
 
 		// packet found, relay from A to B
 		if err := path.EndpointB.UpdateClient(); err != nil {
-			return err
+			return nil, nil, err
 		}
 
 		res, err := path.EndpointB.RecvPacketWithResult(packet)
 		if err != nil {
-			return err
+			return nil, nil, err
 		}
 
 		ack, err := ParseAckFromEvents(res.GetEvents())
 		if err != nil {
-			return err
+			return res, ack, err
 		}
 
-		return path.EndpointA.AcknowledgePacket(packet, ack)
+		return res, ack, path.EndpointA.AcknowledgePacket(packet, ack)
```

Comment:
I would prefer to split this up now that we disabled the linter which changed this (same for below)
```go
if err := path.EndpointA.AcknowledgePacket(packet, ack); err != nil {
    return nil, nil, err
}

return res, ack, nil
```